### ISSUE TITLE
[monitoring v2] `rancher-monitoring-kube-controller-manager` prometheus target is down; pod `rancher-monitoring-admission-patch` fails, causing monitoring install to fail on k8s 1.22

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#35571]
**Rancher Server Setup**
- Rancher version: v2.6-head (96e2438)
- Installation option (Docker install/Helm Chart): Helm 
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): rke1
- Proxy/Cert Details:self signed

**Information about the Cluster**
- Kubernetes version: 1.22.3
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): provider: aws

**Describe the bug**

Monitoring v2 install fails -- on further investigation, most of the install completes but the pod `rancher-monitoring-admission-patch` fails and times out. 

**To Reproduce**
 
* deploy a cluster on 1.22
* deploy monitoring v2
  * install will fail
* while install is happening, go to pods -> `rancher-monitoring-admission-patch` and the logs will show

**Result**
* while install is happening, go to pods -> `rancher-monitoring-admission-patch` and the logs will show the following:
```
W1117 18:45:42.953687 1 client_config.go:608] Neither --kubeconfig nor --master was specified. Using the inClusterConfig. This might not work.
{"level":"info","msg":"patching webhook configurations 'rancher-monitoring-admission' mutating=true, validating=true, failurePolicy=Fail","source":"k8s/k8s.go:39","time":"2021-11-17T18:45:42Z"}
{"err":"the server could not find the requested resource","level":"fatal","msg":"failed getting validating webhook","source":"k8s/k8s.go:48","time":"2021-11-17T18:45:42Z"}
```

**Expected Result**

all parts of monitoring v2/prometheus targets should work on 1.22

**Screenshots**

<img width="1698" alt="Screen Shot 2021-11-17 at 11 36 42 AM" src="https://user-images.githubusercontent.com/16691014/142270509-1b066f7d-565c-406c-90a7-765a46404c7d.png">

**Additional context**

* graphs (i.e. on the cluster page) are displaying and showing data through grafana, so I'm not sure what this could be breaking. There is a [separate bug](https://github.com/rancher/dashboard/issues/4581) blocking other parts of monitoring on this version of 2.6, so I can't do too much other testing at the moment. 
* this is currently specific to 1.22, as 1.21 does not have an issue with the kube-controller target.

**Comment by jiaqiluo** [comment]
**TLDR: It is a known issue in the upstream chart, and the fix is to update both the image and tag used in the pod.** 

More info:
- `jettech/kube-webhook-certgen` is not compatible with 1.22+  [upsteam issue](https://github.com/prometheus-community/helm-charts/issues/1276)
- `jettech/kube-webhook-certgen` is deprecated, the new image and tag are  
```
repository: k8s.gcr.io/ingress-nginx/kube-webhook-certgen
tag: v1.0
```

[upstream PR](https://github.com/prometheus-community/helm-charts/commit/f15adcdcfe6bee8eea6afab5a96e5fa5b415b2d0)

**Comment by jiaqiluo** [comment]
Can be tested in the rancher-monitoring  `100.1.0+up19.0.3`
---
### Root cause
It is a known issue in the upstream chart ( see my previous comment) 

### What was fixed, or what changes have occurred
rebase the monitoring chart to get the fix from the upstream

### Areas or cases that should be tested? What areas could experience regressions? Are the repro steps accurate/minimal?
 
Please check [this commit](https://github.com/rancher/rancher/issues/34577#issuecomment-962151533) for the scope of tests.

**Comment by slickwarren** [comment]
tested on 2.6-head (e322f8f):
* 1.22.3 rke1 cluster -- monitoring v100.1.0+up19.0.3 -- `rancher-monitoring-kube-controller-manager` target is still down
* 1.22.3+rke2r1 (imported) rke2 cluster -- monitoring v100.1.0+up19.0.3 -- `rancher-monitoring-kube-controller-manager` target is still down

it seems that the rebase hasn't fixed the root of this issue for some reason; reopening this ticket

**Comment by brendarearden** [comment]
I was not able to re-create - sending back to test

**Comment by brendarearden** [comment]
I was able to recreate this. 

### Root Cause
https://github.com/kubernetes/kubeadm/issues/2207
2 PRs were done to remove ports from being served insecurely for the kube controller manager starting in k8s 1.22. We are currently still attempting to scrape from the insecure ports our rancher-monitoring chart which is causing the target to be down.

This change occurred for other resources as well https://github.com/kubernetes/kubernetes/issues/91506 so it is expected that kube-scheduler will also have this issue starting in k8s 1.23 https://github.com/kubernetes/kubeadm/issues/2207

### Proposed Solution
The pushprox needs to scrap from the secure port using HTTPS certs from the host for RKE.

**Comment by aiyengar2** [comment]
https://github.com/rancher/rancher/issues/35752

**Comment by brendarearden** [comment]
This is getting bumped to v2.6.4 because v2.6.3 is shipping with k8s 1.22 as experimental. 

To deploy the rancher-monitoring chart successfully in RKE1 /RKE2 1.22.x, you can do the following workaround: 

Ensure that you make the following changes to you values.yaml when installing/upgrading on 1.22.x cluster: 

In the `rkeControllerManager` section in your values.yaml, change the `metricsPort` to the secure `10257` port, and add the `https` values  `enabled: true`, ` insecureSkipVerify: true` and `useServiceAccountCredentials: true` under `clients`. If you are running RKE2 1.22.x, you would need to do these same steps, but under the `rke2ControllerManager` section. 

Example yaml changes in the values:
```
metricsPort: 10257
clients:
  https:
    enabled: true
    insecureSkipVerify: true
    useServiceAccountCredentials: true
```

**Comment by ldbrin** [comment]
> This is getting bumped to v2.6.4 because v2.6.3 is shipping with k8s 1.22 as experimental.
> 
> To deploy the rancher-monitoring chart successfully in RKE1 /RKE2 1.22.x, you can do the following workaround until its fixed in v2.6.4:
> 
> Ensure that you make the following changes to you values.yaml when installing/upgrading on 1.22.x cluster:
> 
> In the `rkeControllerManager` section in your values.yaml, change the `metricsPort` to the secure `10257` port, and add the `https` values `enabled: true`, ` insecureSkipVerify: true` and `useServiceAccountCredentials: true` under `clients`. If you are running RKE2 1.22.x, you would need to do these same steps, but under the `rke2ControllerManager` section.
> 
> Example yaml changes in the values:
> 
> ```
> metricsPort: 10257
> clients:
>   https:
>     enabled: true
>     insecureSkipVerify: true
>     useServiceAccountCredentials: true
> ```

Is there a workaround for k3s?  I am experiencing the same issue with rancher v2.6.3. and k3s v1.22.5+k3s1.

**Comment by loopingz** [comment]
@brendarearden which yaml file are you referring to? Once deployed and failed is it possible to just update a Kubernetes resource to make it work?

**Comment by eumel8** [comment]
100.1.0+up19.0.3 works for me on Kubernetes 1.22.5 / Rancher 2.6.3. THX @jiaqiluo

**Comment by davidnuzik** [comment]
I notice in the UI I don't see KCM data when enabling monitoring as well - probably due to this exact issue.
Strangely I also do not see etcd DB Size as well.

I do see data for both of these when on 1.21 k8s. This is happening on both RKE1 and RKE2 thus I suspect it's a problem with K8S 1.22 specifically (most likely) or rancher/prometheus/grafana not handling something correctly here. I know that's kind of broad 😅 but I think I meant to say I feel this to be k8s API related (deprecated apis issue in 1.22?)

**Comment by ronhorton** [comment]
**PASS** Verified in `2.6-head Commit ID 8ece6ae`

1. created downstream cluster ec2 with k8s 1.22
2. installed monitoring v2 chart
3. observed no error during install
4. after installation, navigate to `rancher//cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/targets`
5. `[serviceMonitor/cattle-monitoring-system/rancher-monitoring-kube-controller-manager/0 (1/1 up)]`
<img width="1022" alt="Screen Shot 2022-03-04 at 11 11 41 AM" src="https://user-images.githubusercontent.com/89212856/156799385-c55940d8-cba1-48af-8fda-304c80e8fdd6.png">
